pkg/inject/report.go

```diff
@@ -19,23 +19,42 @@ type Report struct {
 	UDP                 bool // true if any port in any container has `protocol: UDP`
 	UnsupportedResource bool
 	InjectDisabled      bool
+
+	// Uninjected consists of two boolean flags to indicate if a proxy and
+	// proxy-init containers have been uninjected in this report
+	Uninjected struct {
+		// Proxy is true if a proxy container has been uninjected
+		Proxy bool
+
+		// ProxyInit is true if a proxy-init container has been uninjected
+		ProxyInit bool
+	}
 }
 
 // newReport returns a new Report struct, initialized with the Kind and Name
 // from conf
-func newReport(conf *ResourceConfig) Report {
+func newReport(conf *ResourceConfig) *Report {
 	var name string
-	if m := conf.pod.Meta; m != nil {
+	if m := conf.workload.Meta; m != nil {
+		name = m.Name
+	} else if m := conf.pod.meta; m != nil {
 		name = m.Name
 		if name == "" {
 			name = m.GenerateName
 		}
 	}
 
-	return Report{
+	report := &Report{
 		Kind: strings.ToLower(conf.workload.metaType.Kind),
 		Name: name,
 	}
+	if conf.pod.meta != nil && conf.pod.spec != nil {
+		report.update(conf)
```

Comment:
i really like how we're only calling `report.update(` from `newReport()`. since it's side-effect-y, mind just inlining `update` into this location, and removing the function altogether?